### ISSUE TITLE
Add to Cart button not translating on submission

### ISSUE THREAD

**Issue by BaberAhmed** [magento/magento2#3423]
Hi,

The "Add to Cart" Button text is not translating correctly after the product has been added to the cart. Please see below for screenshots of the issue.

On product page load, the translation for the "Add to Cart" is working as expected and button is translated to "Add to Basket" as shown in the image below. 

![screen shot 2016-02-16 at 11 23 47](https://cloud.githubusercontent.com/assets/13336835/13074959/c9af7a92-d49f-11e5-8009-55e186b2c765.png)

The Product goes through the process of being added to the cart as expected, but when the add to cart process is completed the "Add to Cart" button shows the default string rather than the translation as show in the image below.

![screen shot 2016-02-16 at 11 22 48](https://cloud.githubusercontent.com/assets/13336835/13074947/b3f61468-d49f-11e5-898f-972bedf74000.png)

**Comment by ishakhsuvarov** [comment]
@BaberAhmed Thank you for reporting this issue. We have created an internal ticket `MAGETWO-49560` to fix the problem.

**Comment by moe-kayali** [comment]
I can see the same issue here on Magento 2.0.4. 

How Reproducible: Always.

**Comment by ishakhsuvarov** [comment]
@moe-kayali Sadly, `MAGETWO-49560` is not fixed yet, i'll update this ticket as soon as issue gets fixed.

**Comment by sdzhepa** [comment]
Hello @BaberAhmed. Described issue was fixed in the latest release and was verified on 2.0.7. 
Notice: to apply translation for cacheable pages it is required to flush cache("Translation Cache Type"; System>Cache Management)

**Comment by versdivers** [comment]
I do have this problem still. I have no cache disabled even so the cache problem should not be an issue. It just won't translate..

I have a i18n/nl_NL.csv in my module that is controling and outputting the button which has the next translation in it to test 

```
"Add to Cart","Add to Cartzzz"
Adding...,Adding...
Added,Added,module
```

Then the button should say 'Add to Cartzzz'. But it only says 

![image](https://cloud.githubusercontent.com/assets/18028151/26285496/913522ee-3e51-11e7-8c46-741c28eeee98.png)

**Comment by benjamin-garnier** [comment]
Hello, I've got the same problem, the button Add to cart is pretty stubborn... 
I have made quite a few tests to translate it, putting the words  "Adding...", "Added", "Add to cart", "Add to Cart" in several csv files, in my theme or a module, and sometimes I do have the translation.
It apperas that, on FireFox, the first time I go to the product page, I don't have the translation, but when i reload it and wait a little before trying the Add to cart button, I have the correct translation.
I deleted the js-translation.json files and called setup:static-content:deploy de_DE to regenerate them, but not good enough.
On chrome, It is harder to make the translatin appears. You have to de activate Addblock and such, reload with ctrl+F5 and wait a bit before trying.
It is a weird thing, and I would like to have a decent translation for my website.

**Comment by magento-engcom-team** [comment]
@BaberAhmed, thank you for your report.
The issue is already fixed in develop branch, 2.2.0, 2.1.9

**Comment by rhertogh** [comment]
@magento-engcom-team 
This bug still appears in Magento 2.1.10 in the following browsers

- Firefox 57.0.2 (64-bit)
- Edge 40.15063.674.0
- Internet Explorer 11.786.15063
On Windows 10.

Other javascript translations are working correctly.

In Chrome (63.0.3239.84) the 'add to cart' translation works OK

**Comment by VirtusB** [comment]
I have the same problem with 2.2.1

**Comment by rhertogh** [comment]
@magento-engcom-team please reopen

**Comment by ishakhsuvarov** [comment]
Hi @rhertogh 
Reopening per your request.

**Comment by rhertogh** [comment]
Thanks, current 2.1.x version of magento (we're using 2.1.12) also still shows the same behavior (chrome works, other browsers don't).

**Comment by VirtusB** [comment]
Ours is a little different. "Add to Cart" is translated fine initially, but upon clicking on the button it says "Adding" then "Added", then it resets to "Add to Cart", so after adding a product to the cart none of the translations are effective.

It doesn't matter whether it's Chrome or Firefox

**Comment by evapae** [comment]
It´s still an issue for me... I solved it capturing the text inside the button (so it´s already translated) and then rendering it with the success of the ajax. catalog-add-to-cart.js - yes, I know it´s default magento and I shouldn´t touch it, but it´s not getting fixed any time soon.

In line 61 I get the text (previusly I added an id to the span of the button) 
var buttonName = $('#button-title-js').text();

In line 71 I let the argument travel between functions:
self.ajaxSubmit(form, buttonName);

line78, line 145

ajaxSubmit: function (form, buttonName) {
self.enableAddToCartButton(form, buttonName);

And finally in line 165 
enableAddToCartButton: function (form, buttonName) {

and then render it 174
                var addToCartButtonTextDefault = self.options.addToCartButtonTextDefault || buttonName;

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by TomashKhamlai** [comment]
Thank you for reporting this issue!
I tried to reproduce this issue with language pack from [cubewebsites](https://github.com/cubewebsites/magento2-language-en-gb) and the button was translated all the time. Closing as not reproducible anymore.

**Comment by rudolf81** [comment]
**Solution:**

This issue does _indeed_ appeared to be fixed (community-edition=2.3.4), but the exact same "bug" still manifests, as described by the opening post - _if_ you are not familiar with Magento dev workflow and translations.
This is the translation workflow you need to follow to fix it (or at least worked for me - I'm still learning):

E.g.:
If you have the correct translation string present in your translation csv file, 

_probably located /var/www/html/magento2/vendor/magento/language-xxxxxxx/,_ 

and you've run through a couple of cycles of clearing a numerous caches,

_probably:-
```
cd /var/www/html/magento2/vendor/magento/language-xxxxxx
chown magento_user:www-data *.csv
cd /var/www/html/magento2
php bin/magento indexer:reindex
php bin/magento cache:clean
php bin/magento cache:flush
```

You may find the exact same results as the opening post: add to cart label is correct on loading the product page, but after clicking it, the label reverts.

The reason for this is I believe that that string present on page load comes from the php code which does the correct translation lookup against the dictionary, however the element / ui component gets refreshed from javascript when you click on it.
This time the translation is loaded from a static set of resources which are only processed from the dictionary when you perform this step:

After placing your global dictionary in /var/www/html/magento2/vendor/magento/language-xxxxxxx/, you need to kill your prior static files, here:
`rm -rf /var/www/html/magento2/pub/static/frontend/Magento/luma`
(of whatever theme you are using... default is luma)

And then re-generate them:
```
cd /var/www/html/magento2
php bin/magento setup:static-content:deploy <lang locale of your dictionary e.g. en_GB> -f
```

Followed by some more clearing of the caches:
```
cd /var/www/html/magento2
php bin/magento indexer:reindex
php bin/magento cache:clean
php bin/magento cache:flush
```

Done

That issue is now fixed.

You might notice that many other untranslated strings on the site will suddenly appear correct as well, however, there are still plenty of translation bugs in there waiting for you, of strings that simply are not set up for translation. These could be in php, html or js...
E.g.:
/var/www/html/magento2/vendor/magento/module-review/view/frontend/templates/form.phtml

Have fun hunting them all down! _(Don't forget to try and trigger all manner of validation failures and popups to see if there are any untranslated strings hiding away...)_

**Comment by Aquive** [comment]
5 years later and still present on Magento 2.4.1. In my case it appears only in production mode. The strings are translated ok, until you click the add to cart. It then resets to default translations.

I tried @rudolf81 suggestion without any luck.

**Comment by davirs** [comment]
Magento 2.4.2-p1 problema ainda existe.
